### PR TITLE
RDAP Object is Assumed to be an IP on Country Attribute

### PR DESCRIPTION
**Pull request by MiguelAHM**

### CODE REVIEW

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java [hunk 1]
```diff
@@ -670,13 +670,14 @@ private static void handleMultipleValuesAttributes(final RpslObject rpslObject,
                     rdapObject.setLang(attributeValuesAttribute.next().toString());
                 }
                 case COUNTRY -> {
-                    final Ip ip = (Ip)rdapObject;
-                    if (!StringUtil.isNullOrEmpty(ip.getCountry())){
-                        rdapObject.getRedactedRpslAttrs().addAll(rpslObject.findAttributes(COUNTRY));
-                        break;
+                    if (rdapObject instanceof final Ip ip) {
+                        if (!StringUtil.isNullOrEmpty(ip.getCountry())) {
```
Comment (maggarwal13) [inline whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java:674]: If its only set in IP, can you call this method from createIp in that case you do not need to type cast or use instance of? Similarly for Lang attribute call it from createEnitity and CreateIp only

**Comment by eshryane** [comment]
I updated the title to make it clearer, both for the community and ourselves in the future.

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapServiceTestIntegration.java [hunk 2]
```diff
@@ -374,6 +374,7 @@ public void lookup_org_single_language_codes() {
                 "geoloc:         52.375599 4.899902\n" +
                 "language:       DK\n" +
                 "admin-c:        TP1-TEST\n" +
+                "country:        NL\n"+
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapServiceTestIntegration.java:377]: Can you also add a test to `lookup_org_single_language_codes` and `lookup_org_multiple_language_codes` that country is set properly in the organisation `Entity` response

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java (hunk 1 above)
Comment (eshryane) [reply whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java:674]: Could you create a `mapRedactions` method specifically for Ip ?

Comment (eshryane) [reply whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java:674]: Although then `language` is common to both IP and Organisation - maybe a separate method for that also, but call it with an Entity / RDAP object?

Comment (MiguelAHM) [reply whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java]: Pair programming:
We use just one method with all the logic and we call that method from IPs related logic and Entity related logic.
If in the future this is applied to all the object is very easy to change to the main method

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapServiceTestIntegration.java (hunk 2 above)
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapServiceTestIntegration.java:377]: There is already one test for lookup_org_multiple_language_codes 
I will create lookup_org_single_language_codes_no_redaction

Comment (eshryane) [reply whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapServiceTestIntegration.java:377]: "country" is only specified in IP and aut-num objects, not in entities, so we can't test for it there.

**Review by eshryane — APPROVED** [review]

**Review by maggarwal13 — APPROVED** [review]

**Comment by MiguelAHM** [comment]
Test ok
https://gitlab.ripe.net/swe-database-team/whois/-/pipelines/457698

**Pull request merged**